### PR TITLE
Expose a `Synced` indication on Delta ADSC

### PR DESCRIPTION
**Pull request by howardjohn**
This mirrors the SotW one where we can show when we have actually gotten
everything we watched

### CODE REVIEW

**Review by bleggett — APPROVED** [review]

File: pkg/adsc/delta.go [hunk 1]
```diff
@@ -262,6 +267,11 @@ func (c *Client) Run(ctx context.Context) {
 	c.runWithReconnects(ctx)
 }
 
+// Synced returns a channel that will close once the client has received all initial watches
+func (c *Client) Synced() <-chan struct{} {
```
Comment (hzxuzhonghu) [inline pkg/adsc/delta.go:271]: Provide HasSynced, then no need synced chan

Comment (howardjohn) [reply pkg/adsc/delta.go:271]: A synced channel is strictly better than has synced. HasSynced requires you to poll, a channel allows you to poll OR wait until its ready

Comment (hzxuzhonghu) [reply pkg/adsc/delta.go:271]: Wondering should align with Informer or not. You are right, this is easier to use.

File: pkg/adsc/delta.go [hunk 2]
```diff
@@ -487,6 +506,15 @@ func (c *Client) handleDeltaResponse(d *discovery.DeltaDiscoveryResponse) error
 	return nil
 }
 
+func (c *Client) markReceived(k resourceKey) {
+	if c.pendingWatches.Contains(k) {
+		c.pendingWatches.Delete(k)
+		if c.pendingWatches.Len() == 0 {
+			close(c.synced)
```
Comment (hzxuzhonghu) [inline pkg/adsc/delta.go:513]: how do you prevent double close?

Comment (hzxuzhonghu) [reply pkg/adsc/delta.go:513]: Think first this is closed, but at another time, `initWatch` is called again

Comment (howardjohn) [reply pkg/adsc/delta.go:513]: initWatch can never be called after it starts, it's something done during creation

**Review by hzxuzhonghu — APPROVED** [review]

**Comment by Mariaortizmarquez** [comment]
________________________________

**Review by jacob-delgado — APPROVED** [review]

**Pull request merged**